app/controllers/logins_controller.rb

```diff
@@ -4,4 +4,32 @@
 # a reasonable way to think about a "login" as a resource which is created
 # and deleted (i.e. 'logging out').
 # 
-# Reading and Updating a login, however, make a little less sense.
\ No newline at end of file
+# Reading and Updating a login, however, make a little less sense.
+
+MyApp.get "/" do
+  erb :"/homepage/homepage"
+end
+
+MyApp.get "/login" do
+  erb :"/logins/login"
+end
+
+MyApp.get "/login_attempt" do
+  @user = User.find_by_email(params[:email])
+  @user.name = params["name"]
+   
+   if @user != nil && @user.password == params[:password] 
+
+    session["user_id"] = @user.id
+
+    erb :"/logins/login_success"
+  else
+    erb :"/logins/login_fail"
+  end
+end
+
+MyApp.get "/logout" do
+  @user = nil
```

Comment:
Does this work? What is `@user` in this controller action? How would you describe this feature in pseudo-code?